### ISSUE TITLE
Metadata Completeness

### ISSUE THREAD

**Issue by NicoledeGreef** [bcgov/MFIN-Data-Catalogue#427]
https://github.com/bcgov/MFIN-Data-Catalogue/labels/Idea

---

## User story

As Data Catalogue Product Manager or other Data Governance professional
I want to to be able to run a report on metadata completeness so that 
the Catalogue team can focus our effort on where it is most needed.

## Describe the idea

Stickie captured in initial Service Design workshops:

"Completeness
of Dataset: how
many attributes
are filled or
empty."

While we do have a report based on incomplete Info Schedule values there is not yet a report that covers general completeness.  We'd need a routine to score MRs on Completeness and group them in a way that has value (by Publication State and Organization for example) .

## Actions

- [ ] Horray! I'm going to be a feature!

**Comment by CraigClark** [comment]
We also need a way to measure that. Typically, if content requires certain fields to be complete, you make the field required and it cannot be published unless those fields have values in them. Right now, that is the following fields:

- Summary
- Visibility
- OPR - set when the record is created
- record type - set when the record is created
- personal info
- security

For the rest of them, we would need to determine what fields a record needs to be complete. For example, if the _Location_ field is empty, the record is considered incomplete. Then we need to answer: if the record isn't considered complete without a location, why are we allowing a partial record to be published?

What we have done with the information schedule is unusual. Required fields are the way you normally make sure content is complete.

**Comment by NicoledeGreef** [comment]
I get what you're saying in terms of Required fields but given our current state of data management maturity we have to allow for a minimum in terms of what is Required given that this is new territory for many. We will revisit what is required in order to publish MRs some months out from launch. In the meantime, it might be useful to know that 90 of 121 records are missing a given field value.

That said, we can get at empty fields for Published records via Search (Search for all and export all records to a file). Admins and IM Specialists will be able to search and see records across Orgs, and Managers and Editors will have the same but limited to 
their Orgs. Export XLSX file contains:

title
author
primary_responsibility_org
description
series
created_date
modified_date
data_quality_issues
data_set_historical_change
security_classification
critical_information
authoritative_info
source_system
data_set_type
information_schedule
columns


As an Admin or IM Specialist, or Manager, I might be interested to survey Draft or Unpublished records as well,
 Would it be possible to extend the "Metadata management" results export to include the fields shown above?

In other words, I think the MVP is to allow for a dump that includes State (Draft, Published, Unpublished) or allows a filter.  It's doesn't have to be a pretty report- CSV and/or XLSX format will do.

**Comment by mjmcclung** [comment]
Does the visibility/security get tricky when draft and unpublished records are included?  Right now these are excluded from Search results (a point of confusion, but that's another issue).

**Comment by CraigClark** [comment]
@mjmcclung Only published records are available to search and reports. The only place you will ever see a draft record is on the manage page and in your drafts on the dashboard. You only see unpublished records on the manage page.

**Comment by CraigClark** [comment]
@mjmcclung I'm not sure what you mean by excluding drafts from search being a point of confusion.

**Comment by mjmcclung** [comment]
> @mjmcclung Only published records are available to search and reports. The only place you will ever see a draft record is on the manage page and in your drafts on the dashboard. You only see unpublished records on the manage page.

So there is no point in including State as Nicole suggests as it would only include 'Published' states anyway?

**Comment by mjmcclung** [comment]
> @mjmcclung I'm not sure what you mean by excluding drafts from search being a point of confusion.

I created a draft metadata record and then went to 'find' it via search and it didn't appear (after creating it the page reloads and there is no indication of how to access it to edit it - see #385).  It took me a long time to figure out that I had to scroll down the page to the Draft metadata records section.  I didn't even know about the 'Manage' tab until Christa pointed it out to me.

**Comment by NicoledeGreef** [comment]
@mjmcclung I am suggesting in [my last comment](https://github.com/bcgov/MFIN-Data-Catalogue/issues/427#issuecomment-2046170144) that users with the **Metadata record management** page (i.e. the Manage tab in the Dashboard) can already query according to State. If we extend what gets exported it would make it possible to assess metadata completeness.

**Comment by NicoledeGreef** [comment]
We are closing this one as the Report Builder screen will cover off the requirement. #395
